src/textual/app.py

```diff
@@ -2833,8 +2867,10 @@ async def _close_all(self) -> None:
 
         # Close pre-defined screens.
         for screen in self.SCREENS.values():
-            if isinstance(screen, Screen) and screen._running:
-                await self._prune(screen)
+            # This is now dead code, since instances cannot be in SCREENS.
```

Comment:
I think we can remove this loop entirely, since we prevent instances going in to SCREENS.